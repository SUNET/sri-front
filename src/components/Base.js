import React from "react";
import { Route, Switch } from "react-router-dom";

import SearchContainer from "../containers/Search";
import { ContactDetails } from "./Contact";

class Base extends React.Component {
    render() {
        return (
            <main className="container">
                <Switch>
                    <Route path="/contacts" component={SearchContainer} />
                    <Route
                        path="/contact/:contactId"
                        component={ContactDetails}
                    />
                    <Route path="/roles" component={SearchContainer} />
                </Switch>
            </main>
        );
    }
}

export default Base;
