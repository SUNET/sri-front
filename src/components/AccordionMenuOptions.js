import React from "react";
import { Nav } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import ToggleSection, { ToggleHeading, TogglePanel } from "../components/ToggleSection";

import "../style/AccordionMenuOptions.scss";

class AccordionMenuOptions extends React.Component {
    renderHeader(option, index) {
        return (
            <ToggleHeading key={`${index}`}>
                <div className="accordion-menu-options__header">{option.header}</div>
            </ToggleHeading>
        );
    }
    renderCollapseList(subOptions, optionIndex) {
        return (
            <TogglePanel>
                {subOptions.map((subOption, subOpIndex) => {
                    return (
                        <div key={`option_menu-${optionIndex}-${subOpIndex}`} className="accordion-menu-options__link">
                            <Nav.Link
                                as={NavLink}
                                activeClassName="active"
                                to={subOption.route}
                                onClick={() => this.props.onSelectedOption()}
                            >
                                <i className={`accordion-menu-options__link__icon ${subOption.iconClass}`}></i>
                                <span className="accordion-menu-options__link__name">{subOption.name}</span>
                            </Nav.Link>
                        </div>
                    );
                })}
            </TogglePanel>
        );
    }
    render() {
        const { data } = this.props;

        return (
            <div className="accordion-menu-options">
                {data.map((option, optionIndex) => {
                    return (
                        <ToggleSection key={`option_menu-${optionIndex}`}>
                            {this.renderHeader(option, optionIndex)}
                            {this.renderCollapseList(option.subOptions, optionIndex)}
                        </ToggleSection>
                    );
                })}
            </div>
        );
    }
}

export default AccordionMenuOptions;
