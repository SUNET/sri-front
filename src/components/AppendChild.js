import React from 'react';
import { Field } from 'redux-form';
import { withTranslation } from 'react-i18next';
import FieldInput from './FieldInput';

class AppendChild extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      fields: [{ name: '' }],
    };
  }

  componentWillMount() {}

  handleFieldNameChange = (index) => (event) => {
    const newShareholders = this.state.fields.map((field, sindex) => {
      if (index !== sindex) return field;
      return { ...field, name: event.target.value };
    });

    this.setState({ fields: newShareholders });
  };

  handleAddFild = (event) => {
    event.preventDefault();
    this.setState({
      fields: this.state.fields.concat([{ name: '' }]),
    });
  };

  render() {
    let { t } = this.props;
    return (
      <div>
        {this.state.fields.map((field, index) => (
          <Field
            component={FieldInput}
            type="text"
            placeholder={`${field.name}${index + 1}`}
            value={field.name}
            onChange={this.handleFieldNameChange(index)}
          />
        ))}
        <button className="btn btn-add outline" onClick={(e) => this.handleAddField(e)}>
          <span>{t('actions/add-new')}</span>
        </button>
      </div>
    );
  }
}

export default withTranslation()(AppendChild);
