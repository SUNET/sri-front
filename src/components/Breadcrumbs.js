import React from 'react';
import Breadcrumb from 'react-bootstrap/Breadcrumb';
import PropTypes from 'prop-types';
import Routes from '../Routes';

function generatePathsArray(pathString, isDetailsEntity, entityDetails) {
  return pathString.split('/').map((pathElement, index, array) => {
    const mathIdCurrentDetailsInLocationPath = pathString.includes(entityDetails.id);
    const isLastPath = index === array.length - 1;
    const isActive = isLastPath;
    let pathTo = `/${pathElement}`;
    let name = Routes[pathTo];

    if (typeof Routes[pathTo] === 'undefined') {
      pathTo = array.slice(0, index + 1).join('/');
      name = isLastPath && isDetailsEntity && mathIdCurrentDetailsInLocationPath ? entityDetails.name : Routes[pathTo];
    }
    return {
      pathTo,
      name,
      isActive,
    };
  });
}

const Breadcrumbs = ({ history, isDetailsEntity, entityDetails }) => {
  const { pathname } = history.location;
  const routeSteps = generatePathsArray(pathname, isDetailsEntity, entityDetails);
  return (
    <div className="breadcrumbs">
      <Breadcrumb>
        {routeSteps.map(({ pathTo, name, isActive }, index) => {
          return (
            <Breadcrumb.Item active={isActive} key={`${index * Math.random()}`} href={pathTo}>
              {name}
            </Breadcrumb.Item>
          );
        })}
      </Breadcrumb>
    </div>
  );
};

Breadcrumbs.propsTypes = {
  history: PropTypes.object.isRequired,
  isDetailsEntity: PropTypes.bool.isRequired,
  entityDetails: PropTypes.bool.isRequired,
};

export default Breadcrumbs;
