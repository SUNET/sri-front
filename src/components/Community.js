import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { withTranslation } from "react-i18next";

import SearchOrganizationContainer from "../containers/organization/SearchOrganization";
import SearchContactContainer from "../containers/contact/SearchContact";
import SearchGroupContainer from "../containers/group/SearchGroup";
// import SubMenu from "./SubMenu";
// import SubMenuActions from "./SubMenuActions";

import "../style/Footer.scss";

class Community extends React.Component {
    render() {
        return (
            <Switch>
                <Redirect exact from="/community" to="/community/organizations" />
                <Route path="/community/organizations" component={SearchOrganizationContainer} />
                <Route path="/community/contacts" component={SearchContactContainer} />
                <Route path="/community/groups" component={SearchGroupContainer} />
            </Switch>
        );
    }
}

export default withTranslation()(Community);
