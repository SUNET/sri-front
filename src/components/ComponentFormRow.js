import React from "react";
import { withTranslation } from "react-i18next";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPen, faTrash } from "@fortawesome/free-solid-svg-icons";

class ComponentFormRow extends React.PureComponent {
    constructor() {
        super();

        this.state = {
            is_saved: false,
            is_editing: false,
            is_new: true
        };
    }

    saveRow = (event) => {
        this.setState({ is_saved: true, is_editing: false, is_new: false });
    };

    editRow = (event) => {
        this.setState({ is_saved: false, is_editing: true, is_new: false });
    };

    render() {
        let { t } = this.props;
        return (
            <div>
                {this.props.children(this.state.is_editing && this.state.is_new)}
                {this.props.editable ? (
                    <div className="col-actions">
                        {!this.state.is_editing && !this.state.is_new ? (
                            <div>
                                <FontAwesomeIcon
                                    icon={faTrash}
                                    onClick={() => this.props.fields.remove(this.props.index)}
                                />
                                <FontAwesomeIcon icon={faPen} onClick={(e) => this.editRow(e)} />
                            </div>
                        ) : (
                            <div className="ok-check" onClick={(e) => this.saveRow(e)}>
                                {t("actions.save")}
                            </div>
                        )}
                    </div>
                ) : (
                    <div></div>
                )}
            </div>
        );
    }
}

export default withTranslation()(ComponentFormRow);
