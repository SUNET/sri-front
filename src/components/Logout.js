import React, { Component } from "react";
import { API_HOST } from "../config.js";
import Cookies from "js-cookie";
import { postRequest } from "../sagas/common";

import "../style/Splash.scss";

class Logout extends Component {
    logout = () => {
        Cookies.remove("JWT", { domain: '.localenv.loc' });
        window.location.replace(API_HOST + "/logout");
    };

    render() {
        return <i className="icon-logout" onClick={this.logout} title="Logout"></i>;
    }
}

export default Logout;
