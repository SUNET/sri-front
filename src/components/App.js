import React, { Component } from "react";
import PropTypes from "prop-types";
import { ConnectedRouter } from "connected-react-router";
import Breadcrumbs from "react-router-dynamic-breadcrumbs";
import { Route, Switch } from "react-router-dom";

import FetchingContext from "../components/FetchingContext";
import SplashContainer from "../containers/Splash";
import NotifyContainer from "../containers/Notify";
import TopHeaderContainer from "../containers/TopHeader";
import BaseContainer from "../containers/Base";
import FooterContainer from "../containers/Footer";

import { Row, Col } from "react-bootstrap";
import SideNavNetwork from "./SideNavNetwork";
import SideNavCommunity from "./SideNavCommunity";
import SubMenuActions from "./SubMenuActions";

import Routes from "../Routes";

import "bootstrap/scss/bootstrap.scss";
import "../style/App.scss";
import "../style/Breadcrumbs.scss";
import "../style/SRIButton.scss";

import { history } from "../store";

class App extends Component {
    render() {
        return (
            <FetchingContext.Provider value={this.props.is_fetching}>
                <div className="App container">
                    <ConnectedRouter history={history}>
                        <Row>
                            <Col className="px-0">
                                <TopHeaderContainer />
                            </Col>
                        </Row>
                        <Row>
                            <SplashContainer />
                            <Switch>
                                <Route path="/network" component={SideNavNetwork} />
                                <Route path="/community" component={SideNavCommunity} />
                            </Switch>
                            <Col sm={10} className="fixed-adaptative">
                                <NotifyContainer />
                                <Row className="mt-4">
                                    <Col>
                                        <Breadcrumbs mappedRoutes={Routes} />
                                    </Col>
                                    <Col className="text-right">
                                        <SubMenuActions />
                                    </Col>
                                </Row>
                                {this.props.is_app_loaded && <BaseContainer />}
                            </Col>
                        </Row>
                        <Row>
                            <Col className="px-0">
                                <FooterContainer />
                            </Col>
                        </Row>
                    </ConnectedRouter>
                </div>
            </FetchingContext.Provider>
        );
    }
}

App.propTypes = {
    is_fetching: PropTypes.bool,
    is_app_loaded: PropTypes.bool,
};

export default App;
