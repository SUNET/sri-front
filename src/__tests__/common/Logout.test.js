import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cookies from 'js-cookie';
import { Logout } from '../../components/Logout';

let LogoutWrapper;
let LogoutComponent;

const CreateLogoutWrapper = () => {
  return (
    <div>
      <Logout />
    </div>
  );
};

const mountLogoutWrapper = (props) => mount(<CreateLogoutWrapper {...props} />);

beforeEach(() => {
  jest.clearAllMocks();
  LogoutWrapper = mountLogoutWrapper();
  LogoutComponent = LogoutWrapper.find('Logout');
  delete window.location;
  window.location = { replace: jest.fn() };
  Cookies.remove = jest.fn();
});
afterEach(() => {
  LogoutWrapper.unmount();
});

describe('Logout', () => {
  it('Renders correctly', () => {
    expect(toJson(LogoutWrapper)).toMatchSnapshot();
    expect(LogoutComponent.find('.icon-logout')).toHaveLength(1);
  });
});

describe('Actions', () => {
  it('Click in Logout', () => {
    LogoutComponent.simulate('click');
    expect(window.location.replace).toHaveBeenCalled();
    expect(Cookies.remove).toHaveBeenCalled();
  });
});
