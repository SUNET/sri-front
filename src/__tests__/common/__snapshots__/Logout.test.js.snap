// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Renders correctly 1`] = `
<CreateLogoutWrapper>
  <div>
    <Logout>
      <i
        className="icon-logout"
        onClick={[Function]}
        title="Logout"
      />
    </Logout>
  </div>
</CreateLogoutWrapper>
`;
